go-controller/pkg/persistentips/allocator.go

```diff
@@ -169,11 +169,11 @@ func (icr *IPAMClaimReconciler) Sync(objs []interface{}, ipAllocator IPAllocator
 		if err != nil {
 			return fmt.Errorf("failed at parsing IP when allocating persistent IPs: %w", err)
 		}
-		ips = append(ips, ipnets...)
-	}
-	if len(ips) > 0 {
-		if err := ipAllocator.AllocateIPs(ips); err != nil && !ipam.IsErrAllocated(err) {
-			return fmt.Errorf("failed allocating persistent ips: %w", err)
+
+		if len(ipnets) != 0 {
+			if err := ipAllocator.AllocateIPs(ipnets); err != nil && !ipam.IsErrAllocated(err) {
+				return fmt.Errorf("failed allocating persistent ips: %w", err)
```

Comment:
Let's refernece the function so we differenciate with normal allocations.
```suggestion
				return fmt.Errorf("failed syncing persistent ips: %w", err)
```

Reply:
I don't follow. AFAIU, this log is not repeated anywhere. 

Hence, differentiating it is a no-brainer.

Reply:
indeed it exists just here

Reply:
It's just that this is not "allocating" since there were already assigned, this is syncing them, to make even more clear even if this is the only place.

Reply:
We swallow that already allocated error ...

Not that I care enough to fight for preserving the log.

Reply:
changed in the last push to syncing